cmd/upgrade/cluster/cmd.go

```diff
@@ -428,6 +430,31 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// check if the cluster upgrade requires gate agreements
+	gates, err := ocmClient.GetMissingGateAgreements(cluster.ID(), upgradePolicy)
+	if err != nil {
+		reporter.Errorf("Failed to check for missing gate agreements upgrade for cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	for _, gate := range gates {
+		reporter.Infof("Gate: %v", gate)
+		if !gate.STSOnly() {
+			// for non sts gates we require user agreement
+			if !confirm.Prompt(true,
+				"I acknowledge that my workloads are no longer using removed APIs. (Background:'%s)",
+				gate.Description()) {
+				os.Exit(0)
```

Comment:
The UI here should also include the `documentation_link` from the json payload. The link has important help info for the customer